lib/id3c/cli/command/redcap_det.py

```diff
@@ -111,23 +118,41 @@ def generate(record_ids: List[str], project_id: int, token_name: str, since_date
         ids = record_ids or None,
         raw = True)
 
+    if events:
+        LOG.debug(f"Producing DET notifications for the following events: {events}")
+        assert_known_attribute_value(project, 'events', events)
+    else:
+        LOG.debug(f"Producing DET notifications for all events ({project.events})")
+        events = project.events
+
     if instruments:
         LOG.debug(f"Producing DET notifications for the following {'instruments' if include_incomplete else 'complete instruments'}: {instruments}")
+        assert_known_attribute_value(project, 'instruments', events)
     else:
         LOG.debug(f"Producing DET notifications for all {'instruments' if include_incomplete else 'complete instruments'} ({project.instruments})")
         instruments = project.instruments
 
-    unknown_instruments = set(instruments) - set(project.instruments)
-
-    assert not unknown_instruments, \
-        f"The following --instrument names aren't in the REDCap project: {unknown_instruments}"
 
     for record in records:
+        if record['redcap_event_name'] not in events:
+            continue
+
         for instrument in instruments:
             if include_incomplete or is_complete(instrument, record):
                 print(as_json(create_det_records(project, record, instrument)))
 
 
+def assert_known_attribute_value(project: Project, attribute: str, values: List[str]):
+    """
+    Throws an :class:`AssertionError` if any of the given *values* are not
+    contained in the *attribute* of the given *project*.
+    """
+    unknown_values = set(values) - set(getattr(project, attribute))
+
+    assert not unknown_values, \
+        f"The following {attribute} aren't in the REDCap project: {unknown_values}"
```

Comment:
I ever-so-slightly prefer the previous message text:

> The following --instrument names …

over 

> The following instruments…

because I think the former very clearly ties the problematic values back to the passed in option. With the latter, I wouldn't necessarily know right away it was an error with how I ran the command and not some issue with REDCap or a bug in the code.

Reply:
That makes sense. I added a new argument for the option name which makes the function a bit harder to use, but it doesn't make any assumptions about how to pluralize an option name.